src/error.rs

```diff
@@ -1,14 +1,25 @@
+/// The different set of errors that arroy can encounter.
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
+    /// An internal error mostly related to LMDB or encoding/decoding.
     #[error(transparent)]
     Heed(#[from] heed::Error),
 
+    /// The user is searching for a vector which is not of the right dimensions.
```

Comment:
```suggestion
    /// The user is trying to insert or search for a vector that is not of the right dimensions.
```

The error can also be thrown on the `add_item` method.